### PR TITLE
Noa capability 175213915

### PR DESCRIPTION
**Pull request by ernestoarbitrio**

### CODE REVIEW

File: tests/integration/test_smoothing.py [hunk 1]
```diff
@@ -37,16 +37,16 @@ class DescribeSliceSmoothing(object):
     def it_provides_smoothed_col_pct_for_compatible_cubes(
```
Comment (scanny) [inline tests/integration/test_smoothing.py:37]: Remove lines 4-5 above, now that we're all on Python3 for development we don't need that in tests anymore.

Comment (scanny) [reply tests/integration/test_smoothing.py:37]: Remove lines 12-13 about mnemonics. TR is not used in this module and we all recognize CR for what it is now so these are just noise.

**Review by scanny — APPROVED** [review]
Some changes, after those this LGTM :)

File: src/cr/cube/noa/smoothing.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        """ -> bool, the validity of the window parameter.
+
+        Return last dimension size, if window is greater than the the last dimension
+        size because we cannot have a moving window grater than the number of elements
+        of each column.
+        """
+        total_period = self._base_values.shape[-1]
+        if self._window > total_period or self._window == 0:
+            return False
+        return True
```
[the remark this answers is not in the document]
Comment (ernestoarbitrio) [reply src/cr/cube/noa/smoothing.py]: i'm ok with having this expression inline in `_can_smooth` property. But i think that the condition for an invalid window should be `self._window > self._base_values.shape[-1] or self._window < 2`. When a `window == total_period` the final result is only 1 element (1 col for a 2D or 1 value for a 1D array). Please refer to this example
image

[the remark this answers is not in the document]
Comment (scanny) [reply src/cr/cube/noa/smoothing.py]: The logic I provided was the _positive_ logic, like `if that is True: do smoothing`. The _negative_ logic is just as you say.

**Pull request merged**